client/src/store/auth.js

```diff
@@ -0,0 +1,64 @@
+import axios from "axios";
+
+export const AUTH_REQUEST = "AUTH_REQUEST";
+export const AUTH_SUCCESS = "AUTH_SUCCESS";
+export const AUTH_ERROR = "AUTH_ERROR";
+export const AUTH_LOGOUT = "AUTH_LOGOUT";
+
+const state = {
+    token: localStorage.getItem("user-token") || "",
+    status: "",
+    errorCode: "",
+};
+
+const getters = {
+    isAuthenticated: state => !!state.token,
+    authStatus: state => state.status,
+    error: state => state.errorCode,
+};
+
+const actions = {
+    [AUTH_REQUEST]: ({ commit}, user) => {
+        commit(AUTH_REQUEST);
+        return axios.post("api/user/login", user).then(resp => {
```

Comment:
There is a proxy set up for a dev server (that's the only way I could make it work because of CORS), maybe someone knows how to redirect it on production as well?

Reply:
it should be whitelisted, both local and prod adresses (local only on dev env) on server side, cors have nothing to do with front-end layer